pkg/daemon/update.go

```diff
@@ -315,12 +336,21 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 // and the MCO would just operate on that.  For now we're just doing this to get
 // improved logging.
 type MachineConfigDiff struct {
-	osUpdate bool
-	kargs    bool
-	fips     bool
-	passwd   bool
-	files    bool
-	units    bool
+	osUpdate   bool
+	kargs      bool
+	fips       bool
+	passwd     bool
+	files      bool
+	units      bool
+	kernelType bool
+}
+
+// ValidKernelType returns a valid kernelType, otherwise fallback to default kernel
+func ValidKernelType(kernelType string) string {
+	if kernelType == realtimeKernelType {
+		return realtimeKernelType
+	}
+	return defaultKernelType
```

Comment:
I dont think default should be returned expectedly without a messages saying so - something like: "kernelType %s is not a supported kernel, using defaultKernelType instead," %kernelType

Reply:
to be clear im suggesting this function returns `string, error`